### PR TITLE
added tests to book and category model

### PR DESCRIPTION
**Pull request by irynamekh**
coverage

### CODE REVIEW

File: src/test/java/com/bookstore/service/book/BookServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        assertEquals(VALID_BOOK_RESPONSE_DTO, bookServiceImpl
+                .update(VALID_ID, VALID_BOOK_REQUEST_DTO));
+    }
+
+    @Test
+    @DisplayName("Valid case of the deleteById method run")
+    void deleteById_validBookID_isOk() {
+        doNothing().when(bookRepository).deleteById(VALID_ID);
+        bookServiceImpl.deleteById(VALID_ID);
+        verify(bookRepository, times(1)).deleteById(VALID_ID);
```
Comment (andrii-hoienko) [inline src/test/java/com/bookstore/service/book/BookServiceImplTest.java]: times 1 is by default

File: src/test/java/com/bookstore/service/book/BookServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                List.of(VALID_BOOK_DTO_WITHOUT_CATEGORY_IDS);
+
+        assertEquals(expected, bookServiceImpl.findAllByCategoryId(PAGEABLE, VALID_ID));
+    }
+
+    @Test
+    @DisplayName("Valid case of the getBookById method run")
+    void getBookById_validBookID_returnsValidBook() {
+        when(bookRepository.findById(VALID_ID)).thenReturn(Optional.of(VALID_BOOK));
+
```
Comment (andrii-hoienko) [inline src/test/java/com/bookstore/service/book/BookServiceImplTest.java:152]: Please make a call to the system under test explicitly, it is not clear when you call 'bookServiceImpl' in assert block

File: src/test/resources/database/add_book_and_category.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO book (id, title, author, isbn, price, is_deleted)
+VALUES (1, 'Colony', 'Max Kidruk', '9786176798323', 699.00, false);
+
+INSERT INTO category (id, name, description, is_deleted)
+VALUES (1, 'Science Fiction', 'Imaginative and futuristic concepts such as space exploration and time travel.', false);
+
+INSERT INTO book_category (book_id, category_id)
+VALUES (1, 1);
```
Comment (andrii-hoienko) [inline src/test/resources/database/add_book_and_category.sql]: empty lines

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ! Let`s improve your solution

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**